robottelo/cli/factory.py

```diff
@@ -1702,40 +1704,43 @@ def setup_org_for_a_custom_repo(options=None):
     # Create custom product and repository
     custom_product = make_product({u'organization-id': org_id})
     custom_repo = make_repository({
-        u'url': options.get('url'),
         u'content-type': 'yum',
         u'product-id': custom_product['id'],
+        u'url': options.get('url'),
     })
     # Synchronize custom repository
-    result = Repository.synchronize({'id': custom_repo['id']})
-    if result.return_code != 0:
+    try:
+        Repository.synchronize({'id': custom_repo['id']})
+    except CLICommandExecutionError:
         raise CLIFactoryError('Failed to synchronize repository')
     # Create CV if needed and associate repo with it
     if options.get('content-view-id') is None:
         cv_id = make_content_view({u'organization-id': org_id})['id']
     else:
         cv_id = options['content-view-id']
-    result = ContentView.add_repository({
-        u'id': cv_id,
-        u'repository-id': custom_repo['id'],
-        u'organization-id': org_id,
-    })
-    if result.return_code != 0:
+    try:
+        ContentView.add_repository({
+            u'id': cv_id,
+            u'organization-id': org_id,
+            u'repository-id': custom_repo['id'],
+        })
+    except CLICommandExecutionError:
         raise CLIFactoryError('Failed to add repository to content view')
     # Publish a new version of CV
-    result = ContentView.publish({u'id': cv_id})
-    if result.return_code != 0:
+    try:
+        ContentView.publish({u'id': cv_id})
+    except CLICommandExecutionError:
         raise CLIFactoryError('Failed to publish new version of content view')
     # Get the version id
-    result = ContentView.info({u'id': cv_id})
-    cvv = result.stdout['versions'][-1]
+    cvv = ContentView.info({u'id': cv_id})['versions'][-1]
     # Promote version to next env
-    result = ContentView.version_promote({
-        u'id': cvv['id'],
-        u'to-lifecycle-environment-id': env_id,
-        u'organization-id': org_id,
-    })
-    if result.return_code != 0:
+    try:
+        ContentView.version_promote({
+            u'id': cvv['id'],
+            u'organization-id': org_id,
+            u'to-lifecycle-environment-id': env_id,
+        })
+    except CLICommandExecutionError:
```

Comment:
How about including the exception in this error message?

```python
except CLICommandExecutionError as err:
    raise ...  # include err in new exception
```

Same question applies to rest of module. I'll refrain from commenting more.